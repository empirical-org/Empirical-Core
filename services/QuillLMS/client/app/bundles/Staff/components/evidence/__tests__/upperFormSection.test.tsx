import { shallow } from 'enzyme';
import * as React from 'react';

import UpperFormSection from '../configureSettings/upperFormSection';

const mockProps = {
  activity: {},
  activityTitle: '',
  activityNotes: '',
  activityFlag: {},
  errors: [],
  formErrorsPresent: false,
  handleSetActivityFlag: jest.fn(),
  handleSetActivityNotes: jest.fn(),
  handleSetActivityTitle: jest.fn(),
  handleSubmitActivity: jest.fn(),
  parentActivityId: 1,
  requestErrors: [],
  showErrorsContainer: false
}

describe('UpperFormSection component', () => {
  const container = shallow(<UpperFormSection {...mockProps} />);

  it('should render UpperFormSection', () => {
    expect(container).toMatchSnapshot();
  });
});
