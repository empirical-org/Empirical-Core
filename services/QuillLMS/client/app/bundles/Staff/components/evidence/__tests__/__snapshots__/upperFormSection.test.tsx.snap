// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpperFormSection component should render UpperFormSection 1`] = `
<Fragment>
  <Input
    autoComplete="on"
    className="title-input"
    handleChange={[MockFunction]}
    label="Activity Name"
    value=""
  />
  <Input
    autoComplete="on"
    className="name-input"
    handleChange={[MockFunction]}
    label="Internal Name"
    value=""
  />
  <div
    className="button-and-id-container"
  >
    <section
      className="label-status-container"
    >
      <p
        id="label-status-label"
      >
        Parent Activity ID
      </p>
      <p
        id="label-status"
      >
        1
      </p>
    </section>
    <button
      className="quill-button-archived fun primary contained focus-on-light"
      id="activity-submit-button"
      onClick={[MockFunction]}
      type="submit"
    >
      Save
    </button>
  </div>
  <div
    className="dropdowns-container"
  >
    <DropdownInput
      className="flag-input"
      handleChange={[MockFunction]}
      label="Activity Flag"
      options={
        Array [
          Object {
            "label": "alpha",
            "value": "alpha",
          },
          Object {
            "label": "private",
            "value": "private",
          },
          Object {
            "label": "beta",
            "value": "beta",
          },
          Object {
            "label": "gamma",
            "value": "gamma",
          },
          Object {
            "label": "Evidence Beta 1",
            "value": "evidence_beta1",
          },
          Object {
            "label": "Evidence Beta 2",
            "value": "evidence_beta2",
          },
          Object {
            "label": "College Board",
            "value": "college_board",
          },
          Object {
            "label": "production",
            "value": "production",
          },
          Object {
            "label": "archived",
            "value": "archived",
          },
        ]
      }
    />
    <DropdownInput
      className="ai-type-input"
      label="AI Type"
      options={
        Array [
          Object {
            "label": "AutoML",
            "value": "AutoML",
          },
          Object {
            "label": "RAG",
            "value": "RAG",
          },
          Object {
            "label": "GenAI",
            "value": "GenAI",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
